pgmpy/readwrite/XMLBIF.py

```diff
@@ -182,3 +184,225 @@ def get_model(self):
+        -------
+        >>> writer = XMLBIFWriter(model)
+        >>> writer.get_states()
+        {'dog-out': [<Element OUTCOME at 0x7ffbabfcdec8>, <Element OUTCOME at 0x7ffbabfcdf08>],
+         'family-out': [<Element OUTCOME at 0x7ffbabfd4108>, <Element OUTCOME at 0x7ffbabfd4148>],
+         'bowel-problem': [<Element OUTCOME at 0x7ffbabfd4088>, <Element OUTCOME at 0x7ffbabfd40c8>],
+         'hear-bark': [<Element OUTCOME at 0x7ffbabfcdf48>, <Element OUTCOME at 0x7ffbabfcdf88>],
+         'light-on': [<Element OUTCOME at 0x7ffbabfcdfc8>, <Element OUTCOME at 0x7ffbabfd4048>]}
+        """
+        outcome_tag = {}
+        cpds = self.model.get_cpds()
+        for cpd in cpds:
+            var = cpd.variable
+            outcome_tag[var] = []
+            for state in cpd.variables[var]:
+                state_tag = etree.SubElement(self.variables[var], "OUTCOME")
+                state_tag.text = state
+                outcome_tag[var].append(state_tag)
+        return outcome_tag
+
+    def get_properties(self):
+        """
+        Add property to variables in XMLBIF
+
+        Return
+        ---------------
+        dict: dict of type {variable: property tag}
+
+        Examples
+        -------
+        >>> writer = XMLBIFWriter(model)
+        >>> writer.add_property()
+        {'light-on': <Element PROPERTY at 0x7f7a2ffac1c8>,
+         'family-out': <Element PROPERTY at 0x7f7a2ffac148>,
+         'hear-bark': <Element PROPERTY at 0x7f7a2ffac188>,
+         'bowel-problem': <Element PROPERTY at 0x7f7a2ffac0c8>,
+         'dog-out': <Element PROPERTY at 0x7f7a2ffac108>}
+        """
+        variables = self.model.nodes()
+        property_tag = {}
+        for var in sorted(variables):
+            properties = self.model.node[var]
+            property_tag[var] = etree.SubElement(self.variables[var], "PROPERTY")
+            for prop, val in properties.items():
+                property_tag[var].text = str(prop) + " = " + str(val)
+        return property_tag
+
+    def get_definition(self):
+        """
+        Add Definition to XMLBIF
+
+        Return
+        ---------------
+        dict: dict of type {variable: definition tag}
+
+        Examples
+        -------
+        >>> writer = XMLBIFWriter(model)
+        >>> writer.add_definition()
+        {'hear-bark': <Element DEFINITION at 0x7f1d48977408>,
+         'family-out': <Element DEFINITION at 0x7f1d489773c8>,
+         'dog-out': <Element DEFINITION at 0x7f1d48977388>,
+         'bowel-problem': <Element DEFINITION at 0x7f1d48977348>,
+         'light-on': <Element DEFINITION at 0x7f1d48977448>}
+        """
+        cpds = self.model.get_cpds()
+        cpds.sort(key=lambda x: x.variable)
+        definition_tag = {}
+        for cpd in cpds:
+            definition_tag[cpd.variable] = etree.SubElement(self.network, "DEFINITION")
+            etree.SubElement(definition_tag[cpd.variable], "FOR").text = cpd.variable
+            for child in sorted(cpd.evidence):
```

Comment:
@vivek425ster This will throw an error when there's no evidence variables in the cpd. You can do something like `sorted([] if cpd.evidence == None else cpd.evidence)`